src/20250114_AaveV3Ethereum_GSMsMigrationToGSM4626/AaveV3Ethereum_GSMsMigrationToGSM4626_20250114.sol

```diff
@@ -0,0 +1,252 @@
+
+  bytes32 public constant LIQUIDATOR_ROLE =
+    0x5e17fc5225d4a099df75359ce1f405503ca79498a8dc46a7d583235a0ee45c16; // keccak256('LIQUIDATOR_ROLE')
+  bytes32 public constant SWAP_FREEZER_ROLE =
+    0x6dac4cc0544e34aa1a4ed2862f6de78290e3f18f00fe77179ee8ef34de9dfa24; // keccak245('SWAP_FREEZER_ROLE')
+  uint128 public constant USDC_CAPACITY = 8_000_000 ether;
+  uint128 public constant USDT_CAPACITY = 16_000_000 ether;
+
+  // https://etherscan.io/address/0xFeeb6FE430B7523fEF2a38327241eE7153779535
+  address public constant NEW_GSM_USDC = 0xFeeb6FE430B7523fEF2a38327241eE7153779535;
+  string public constant GSM_USDC_NAME = 'GSM 4626 stataUSDC';
+
+  // https://etherscan.io/address/0x5f614E8f948fe5eE4F20F7eD517aea5C911BFb85
+  address public constant USDC_ORACLE_SWAP_FREEZER = 0x5f614E8f948fe5eE4F20F7eD517aea5C911BFb85;
+
+  // https://etherscan.io/address/0x535b2f7C20B9C83d70e519cf9991578eF9816B7B
+  address public constant NEW_GSM_USDT = 0x535b2f7C20B9C83d70e519cf9991578eF9816B7B;
+  string public constant GSM_USDT_NAME = 'GSM 4626 stataUSDT';
+
+  // https://etherscan.io/address/0xca5792C238FA1A6B7805448b3507193Ab160228E
+  address public constant USDT_ORACLE_SWAP_FREEZER = 0xca5792C238FA1A6B7805448b3507193Ab160228E;
+
+  // https://etherscan.io/address/0x83896a35db4519BD8CcBAF5cF86CCA61b5cfb938
+  address public constant FEE_STRATEGY = 0x83896a35db4519BD8CcBAF5cF86CCA61b5cfb938;
+
+  // https://etherscan.io/address/0x1cDF8879eC8bE012bA959EB515b11008E0cb6323
+  address public constant ROBOT_OPERATOR = 0x1cDF8879eC8bE012bA959EB515b11008E0cb6323;
+  uint96 public constant LINK_AMOUNT_ORACLE_FREEZER_KEEPER = 80 ether;
+  uint96 public constant TOTAL_LINK_AMOUNT_KEEPERS = LINK_AMOUNT_ORACLE_FREEZER_KEEPER * 2; // 2 GSMs
+  uint32 public constant KEEPER_GAS_LIMIT = 150_000;
+
+  uint256 public constant EXISTING_ORACLE_SWAP_FREEZER_USDC =
+    51847655505574639685479937049206026612575185607334297935344449721698948000090;
+  uint256 public constant EXISTING_ORACLE_SWAP_FREEZER_USDT =
+    10951818704514842241879618761243590851930089501471583798009172184131135112965;
+
+  function execute() external {
+    uint256 balanceUsdc = IERC20(AaveV3EthereumAssets.USDC_UNDERLYING).balanceOf(
+      GhoEthereum.GSM_USDC
+    );
+    uint256 balanceUsdt = IERC20(AaveV3EthereumAssets.USDT_UNDERLYING).balanceOf(
+      GhoEthereum.GSM_USDT
+    );
+
+    _seize();
+    _grantAccess();
+    _updateFeeStrategy();
+    _registerOracles();
+    _fund(balanceUsdc, balanceUsdt);
+    _revokeAccess();
+  }
+
+  function _seize() internal {
+    IGsm(GhoEthereum.GSM_USDC).grantRole(LIQUIDATOR_ROLE, GovernanceV3Ethereum.EXECUTOR_LVL_1);
+    IGsm(GhoEthereum.GSM_USDT).grantRole(LIQUIDATOR_ROLE, GovernanceV3Ethereum.EXECUTOR_LVL_1);
+
+    IGsm(GhoEthereum.GSM_USDC).seize();
+    IGsm(GhoEthereum.GSM_USDT).seize();
+  }
+
+  function _grantAccess() internal {
+    // Enroll GSMs as GHO Facilitators
+    IGhoToken(AaveV3EthereumAssets.GHO_UNDERLYING).addFacilitator(
```

Comment:
we should remove old gsm instances as facilitators, or at least set bucket capacity to 0